feat: add EventJsonLd component

kenleytomlin opened the pull request:
## Description of Changes:

1. Added EventJsonLd component
2. Made address type an interface so it can be reused
3. Moved buildAddress function into a separate file so it can be reused
4. Updated README.md

garmeeh commented on src/jsonld/event.tsx:
```diff
@@ -0,0 +1,66 @@
+  images?: string[];
+}
+
+const buildLocation = (location: Location) => `
```
Because `location.sameAs` is optional I think the JSON will be invalid when due to the trailing `,` from `buildAddress()` above it. If you move name to be last (and remove `,`)  and give sameAs a trailing comma it should be ok I think. So it would look like: 

```
const buildLocation = (location: Location) => `
  "location": {
    "@type": "Place",
    ${buildAddress(location.address)}
    ${location.sameAs ? `"sameAs": "${location.sameAs}",` : ``}
    "name": "${location.name}"
  },
`;
```

garmeeh commented on src/jsonld/event.tsx:
```diff
@@ -0,0 +1,66 @@
+  description,
+  images,
+}) => {
+  const jslonld = `{
```
I think we will run into another issue here we trailing commas 😅 

If we move things around we should be able to ensure it will be ok even when all optional fields are not set. Something like this should work:
```
"@context": "http://schema.org",
    "@type": "Event",
    "startDate": "${startDate}",
    "endDate": "${endDate}",
    ${buildLocation(location)}
    ${images ? `"image":${formatIfArray(images)},` : ``}
    ${url ? `"url": "${url}",` : ``}
    ${description ? `"description": "${description}",` : ``}
    "name": "${name}"
```

garmeeh commented on README.md:
````diff
@@ -1367,7 +1368,55 @@ export default () => (
 );
 ```
 
-[Google Docs for Social Profile](https://developers.google.com/search/docs/data-types/social-profile)
````
Could add this back in and move it to be at the end of [this section](https://github.com/garmeeh/next-seo#social-profile) just before the News heading.

garmeeh commented:
@all-contributors please add @kenleytomlin for code, doc, test

kenleytomlin commented:
@garmeeh I think I have addressed all of your comments, if there is anymore feedback please let me know. Thanks for reviewing and adding me to the contributors.
